compiler/rustc_middle/src/ty/instance.rs

```diff
@@ -657,7 +657,7 @@ impl<'tcx> Instance<'tcx> {
         T: TypeFoldable<TyCtxt<'tcx>> + Clone,
     {
         if let Some(substs) = self.substs_for_mir_body() {
-            tcx.try_subst_and_normalize_erasing_regions(substs, param_env, v)
+            tcx.try_subst_and_normalize_erasing_regions(substs, param_env, ty::EarlyBinder(v))
```

Comment:
Should these functions also take an `EarlyBinder`?

Reply:
Hmm yeah it seems like they probably should. I tried this in a new commit, please let me know what you think.

Something I'm generally wondering about is how far to "bubble up" these `EarlyBinder` changes, like `subst_mir_and_normalize_erasing_regions` doesn't get called too much directly but it's wrapped by some other methods that get called more, and I just left those as `T` instead of changing them to take `EarlyBinder<T>`. Happy to start changing those as well, just wanted to ask first

Reply:
I can look later today. In the most ideal world, we bubble up EarlyBinder creation to where we create the params and ReEarlyBound. For this PR, it'll likely be some balance of "makes sense" and "small enough"